modules/apps/27-interchain-accounts/types/genesis.go

```diff
@@ -30,6 +49,41 @@ func NewControllerGenesisState(channels []ActiveChannel, accounts []RegisteredIn
 	}
 }
 
+// Validate performs basic validation of the ControllerGenesisState
+func (gs ControllerGenesisState) Validate() error {
+	for _, ch := range gs.ActiveChannels {
+		if err := host.ChannelIdentifierValidator(ch.ChannelId); err != nil {
+			return err
+		}
+
+		if err := host.PortIdentifierValidator(ch.PortId); err != nil {
+			return err
+		}
+	}
+
+	for _, acc := range gs.InterchainAccounts {
+		if err := host.PortIdentifierValidator(acc.PortId); err != nil {
+			return err
+		}
+
+		if !IsValidAddr(acc.AccountAddress) || len(acc.AccountAddress) == 0 || len(acc.AccountAddress) > DefaultMaxAddrLength {
```

Comment:
Maybe we should make a `InterchainAccAddressValidator` function?

Reply:
Done 👍